### PR TITLE
fix(Member): User property reference

### PR DESCRIPTION
**Pull request by ghost**
The user's values do not get updated in member objects. This pr fixes this issue.

### CODE REVIEW

File: lib/structures/Member.js [hunk 1]
```diff
@@ -57,7 +57,11 @@ class Member extends Base {
                 throw new Error("User associated with Member not found: " + data.id);
             }
         } else if(data.user) {
-            this.user = new User(data.user, client);
+            if(!client) {
+                this.user = new User(data.user, client);
```
Comment (bsian03) [inline lib/structures/Member.js]: Useless client

File: lib/structures/Member.js [hunk 2]
```diff
@@ -57,7 +57,11 @@ class Member extends Base {
...
+            } else {
+                this.user = client.users.add(data.user, client);
```
Comment (bsian03) [inline lib/structures/Member.js]: Can probably update instead of add

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: lib/structures/Member.js (hunk 1 above)
Comment (ghost) [reply lib/structures/Member.js]: Oh okay, i just thought that the client property may not always be passed.

File: lib/structures/Member.js [hunk 3]
```diff
@@ -57,7 +57,7 @@ class Member extends Base {
                 throw new Error("User associated with Member not found: " + data.id);
             }
         } else if(data.user) {
-            this.user = new User(data.user, client);
+            this.user = client.users.update(data.user, client);
```
Comment (bsian03) [inline lib/structures/Member.js]: Client is still not guaranteed to exist at this location

**Review by bsian03 — APPROVED** [review]

**Pull request merged**